packages/widget.api/src/config.ts

```diff
@@ -26,9 +26,15 @@ interface ArrayParameterDefinition<S extends ParameterValue.PrimitiveType> {
   displayName: string;
   subType: S;
 }
+interface ObjectSetParameterDefinition<T extends ObjectType> {
+  type: "objectSet";
+  displayName: string;
+  objectType: T;
+}
 export type ParameterDefinition =
   | PrimitiveParameterDefinition<ParameterValue.PrimitiveType>
-  | ArrayParameterDefinition<ParameterValue.PrimitiveType>;
+  | ArrayParameterDefinition<ParameterValue.PrimitiveType>
+  | ObjectSetParameterDefinition<ObjectType>;
```

Comment:
Have you thought about what an event with updateParameterIds that includes an object set parameter will do?

Reply:
I've played around with the existing [bidirectional iframe widgets](https://github.com/palantir/workshop-iframe-custom-widget) and they can actually pass a new object set rid up to Workshop which Workshop is able to use.

I think we could allow custom widgets to also do the same, i.e. emit an update event that sets the object set variable in Workshop to a new object set rid value. What are your thoughts?